mm/2s2h/Rando/Logic/Logic.h

```diff
@@ -62,7 +62,9 @@ extern std::map<RandoRegionId, RandoRegion> Regions;
 #define CAN_USE_HUMAN_SWORD (GET_CUR_EQUIP_VALUE(EQUIP_TYPE_SWORD) >= EQUIP_VALUE_SWORD_KOKIRI)
 #define CAN_USE_SWORD (CAN_USE_HUMAN_SWORD || HAS_ITEM(ITEM_SWORD_GREAT_FAIRY) || CAN_BE_DEITY)
 // Be careful here, as some checks require you to play the song as a specific form
-#define CAN_PLAY_SONG(song) (HAS_ITEM(ITEM_OCARINA_OF_TIME) && CHECK_QUEST_ITEM(QUEST_SONG_##song))
+#define CAN_PLAY_SONG(song)                                                   \
+    (HAS_ITEM(ITEM_OCARINA_OF_TIME) && CHECK_QUEST_ITEM(QUEST_SONG_##song) && \
+     Rando::Logic::canPlaySong(Ship_GetOcarinaSongByQuestId(QUEST_SONG_##song)))
```

Comment:
Since only a valid QUEST_SONG can exist in this, I don't think the associated function and map are needed. This can be handled like CAN_USE_ABILITY:

```suggestion
     Rando::Logic::canPlaySong((QUEST_SONG_##song - QUEST_SONG_SONATA) + OCARINA_SONG_SONATA))
```